packages/@ember/-internals/routing/lib/system/route.ts

```diff
@@ -561,76 +561,6 @@ class Route extends EmberObject.extend(ActionHandler, Evented) implements IRoute
     this.trigger('activate', transition);
   }
 
-  /**
-    The `willTransition` action is fired at the beginning of any
-    attempted transition with a `Transition` object as the sole
-    argument. This action can be used for aborting, redirecting,
-    or decorating the transition from the currently active routes.
-
-    A good example is preventing navigation when a form is
-    half-filled out:
-
-    ```app/routes/contact-form.js
-    import Route from '@ember/routing/route';
-    import { action } from '@ember/object';
-
-    export default class ContactFormRoute extends Route {
-      @action
-      willTransition(transition) {
-        if (this.controller.get('userHasEnteredData')) {
-          this.controller.displayNavigationConfirm();
-          transition.abort();
-        }
-      }
-    }
-    ```
-
-    You can also redirect elsewhere by calling
-    `this.transitionTo('elsewhere')` from within `willTransition`.
-    Note that `willTransition` will not be fired for the
-    redirecting `transitionTo`, since `willTransition` doesn't
-    fire when there is already a transition underway. If you want
-    subsequent `willTransition` actions to fire for the redirecting
-    transition, you must first explicitly call
-    `transition.abort()`.
-
-    To allow the `willTransition` event to continue bubbling to the parent
-    route, use `return true;`. When the `willTransition` method has a
-    return value of `true` then the parent route's `willTransition` method
-    will be fired, enabling "bubbling" behavior for the event.
-
-    @event willTransition
-    @param {Transition} transition
-    @since 1.0.0
-    @public
-  */
-
-  /**
-    The `didTransition` action is fired after a transition has
-    successfully been completed. This occurs after the normal model
-    hooks (`beforeModel`, `model`, `afterModel`, `setupController`)
-    have resolved. The `didTransition` action has no arguments,
-    however, it can be useful for tracking page views or resetting
-    state on the controller.
-
-    ```app/routes/login.js
-    import Route from '@ember/routing/route';
-    import { action } from '@ember/object';
-
-    export default class LoginRoute extends Route {
-      @action
-      didTransition() {
-        this.controller.get('errors.base').clear();
-        return true; // Bubble the didTransition event
-      }
-    }
-    ```
-
-    @event didTransition
-    @since 1.2.0
-    @public
-  */
-
```

Comment:
The deprecation guide is not clear, and the code at the bottom of this file, if the hooks on the Route are deprecated as well.